packages/core/strapi/lib/services/entity-service/types/index.d.ts

```diff
@@ -1,14 +1,36 @@
-import type { Attribute, Common, EntityService } from '@strapi/strapi';
+import type { Attribute, Common, Utils } from '@strapi/strapi';
+import type { Result, PaginatedResult } from './result';
+
+import type * as Params from './params';
 
 // TODO: Move params to @strapi/utils (related to convert-query-params)
 export * as Params from './params';
 
+export * from './result';
 export * from './plugin';
 
+type WrapAction = Omit<keyof EntityService, 'wrapParams' | 'wrapResult' | 'emitEvent'>;
+
 export interface EntityService {
+  wrapParams<TContentTypeUID extends Common.UID.ContentType, TParams extends object>(
+    params?: TParams,
+    options?: { uid: TContentTypeUID; action: WrapAction }
+  ): unknown;
+
+  wrapResult<TContentTypeUID extends Common.UID.ContentType>(
+    result: any,
+    options?: { uid: TContentTypeUID; action: WrapAction }
+  ): any;
+
+  emitEvent<TContentTypeUID extends Common.UID.ContentType>(
+    uid: TContentTypeUID,
+    event: Event,
```

Comment:
```suggestion
    event: string,
```
These aren't actual events, they're the names of events. If we really wanted, we could export ALLOWED_WEBHOOK_EVENTS  as const and get the values from that, but I don't think it's worth it.